pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -0,0 +1,282 @@
+				old := o.(*extensionsv1alpha1.Network)
+				return validation.ValidateNetworkUpdate(new, old)
+			}})
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "operatingsystemconfigs"},
+		artifact{
+			newEntity: func() client.Object { return new(extensionsv1alpha1.OperatingSystemConfig) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.OperatingSystemConfig)
+				return validation.ValidateOperatingSystemConfig(new)
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.OperatingSystemConfig)
+				old := o.(*extensionsv1alpha1.OperatingSystemConfig)
+				return validation.ValidateOperatingSystemConfigUpdate(new, old)
+			}})
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "workers"},
+		artifact{
+			newEntity: func() client.Object { return new(extensionsv1alpha1.Worker) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorker(new)
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Worker)
+				old := o.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorkerUpdate(new, old)
+			}})
+
+	return &h
+}
+
+type handler struct {
+	decoder   *admission.Decoder
+	logger    logr.Logger
+	artifacts map[metav1.GroupVersionResource]artifact
+}
+
+func (h *handler) addResource(gvr metav1.GroupVersionResource, art artifact) {
+	h.artifacts[gvr] = art
+}
+
+var _ admission.Handler = &handler{}
+
+func (h *handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *handler) Handle(ctx context.Context, ar admission.Request) admission.Response {
+	h.logger.Info(fmt.Sprintf("validating resource of type %s for operation %s", ar.Resource, ar.Operation))
+
+	_, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	artifact, ok := h.artifacts[ar.Resource]
+	if !ok {
+		return admission.Allowed("validation not found for the given resource")
+	}
+
+	switch ar.Operation {
+	case admissionv1.Create:
+		return h.handleValidation(ar.Object, ar.OldObject, artifact.newEntity, artifact.validateCreateResource)
+	case admissionv1.Update:
+		return h.handleValidation(ar.Object, ar.OldObject, artifact.newEntity, artifact.validateUpdateResource)
+	default:
+		return admission.Allowed("operation is not CREATE or UPDATE")
+	}
+}
+
+type newEntity func() client.Object
+type validate func(new, old client.Object) field.ErrorList
+
+// artifact servers as a helper to setup the corresponding function.
+type artifact struct {
+	// newEntity is a simple function that creates and returns a new resource.
+	newEntity newEntity
```

Comment:
Nit: I think `newObject` would be a better name here (both for the type and the field)
```suggestion
	newObject newObjectFunc
```